src/js/components/message.jsx

```diff
@@ -57,14 +62,14 @@ class Message extends Component {
     }
 
     render() {
-        const {name, role, avatarUrl, text, accentColor, firstInGroup, lastInGroup, linkColor, type, mediaUrl, sendStatus, clickToRetryText, tapToRetryText} = this.props;
-
-        const actions = this.props.actions.filter((a) => a.type !== 'reply');
+        const {name, role, avatarUrl, text, accentColor, firstInGroup, lastInGroup, linkColor, type, mediaUrl, sendStatus, clickToRetryText, tapToRetryText, locationSendingFailedText} = this.props;
+        const actions = this.props.actions.filter(({type}) => type !== 'reply' && type !== 'locationRequest');
         const hasText = text && text.trim() && text.trim() !== mediaUrl;
         const hasImage = type === 'image';
+        const hasLocation = type === 'location';
         const isAppUser = role === 'appUser';
         const hasActions = actions.length > 0;
-        const lastItem = hasActions ? 'actions' : hasText ? 'text' : null;
+        const lastItem = hasActions ? 'actions' : hasText ? 'text' : hasLocation ? 'location' : null;
```

Comment:
This is for ones that are currently uploading.
If it has text, we are displaying that text as normal, and so the lastItem displayed will in fact be `text`.